From 483bafb4179229714d920de49ba964170c212ef9 Mon Sep 17 00:00:00 2001
From: Weida Hong <wdhongtw@gmail.com>
Date: Thu, 29 Mar 2018 18:37:20 +0800
Subject: [PATCH] Fix assertion related compile failure

static_assert is declared in assert.h and should only be used after c11.
---
 xdelta3/Makefile.am | 2 +-
 xdelta3/xdelta3.h   | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/xdelta3/Makefile.am b/xdelta3/Makefile.am
index 2280083..2d41213 100644
--- a/xdelta3/Makefile.am
+++ b/xdelta3/Makefile.am
@@ -73,7 +73,7 @@ WFLAGS = -Wall -Wshadow -fno-builtin -Wextra -Wsign-compare \
  # -Wno-variadic-macros \
  # -Wno-c++98-compat-pedantic
 
-C_WFLAGS = $(WFLAGS) -pedantic -std=c99
+C_WFLAGS = $(WFLAGS) -pedantic -std=c11
 CXX_WFLAGS = $(WFLAGS) -pedantic -std=c++11
 
 common_CFLAGS = \
diff --git a/xdelta3/xdelta3.h b/xdelta3/xdelta3.h
index 0c1ae0d..7d5e230 100644
--- a/xdelta3/xdelta3.h
+++ b/xdelta3/xdelta3.h
@@ -32,6 +32,7 @@
 #include "config.h"
 #endif
 
+#include <assert.h>
 #include <errno.h>
 #include <stdarg.h>
 #include <stddef.h>
-- 
2.22.0

